### DISCUSSION TITLE
Question: Can ListBox be improved further?

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1426] [Q&A] [answered]
After  https://github.com/BlitterStudio/amiberry/commit/d34f437687331c7a8311a9c06a06a35069297cc5 I've been trying to understand the effect of setSelected(0) ... with default colors..

![ex](https://imgur.com/Pe0q3gq.png)

....with custom theme colors...

![ex2](https://imgur.com/34sMmDz.png)

Eventually I tracked this down to   external/libguisan/src/widgets/listbox.cpp @ line 136

`const auto inactive_color = Color(170, 170, 170);`

This is the highlight color setSelected(0) is using.

In essence, this is equivalent to `gui_theme_selector_inactive=170, 170, 170`  ... the default color here.

To me, this inferred that a construct like  `const auto inactive_color = Color(getInactiveSelectorColor)` is going to work exactly the same with the default color schema. In theory, the highlight color for setSelected(0) would then be more or less appropriate, for other GUI themes as well, and would change depending on the values of `gui_theme_selector_inactive` ...

To test this, my GUI theme part in amiberry.conf  is  `gui_theme_selector_inactive=13, 7, 57`  
--> edit listbox.cpp @ line 136 to be   `const auto inactive_color = Color(13, 7, 57);`
--->recompile amiberry and launch...

![ex3](https://imgur.com/5XuK6An.png)

That's actually pretty good in my book (definitely less of an eye-sore compared to the bright highlighting on dark GUI theme), and to me at least seems a better approach than having setSelected(0) always return a highlight color=170,170,170 ~ it should change in concert with the GUI theme colors ideally.

Is there any way to implement this globally, without hacking Guisan? (I'm thinking the answer is 'no' ;)

TIA

ps: as always, I'm fine with just hacking my local copy to suit =)

**Comment by midwan** [comment, marked as answer]
this has to be changed in guisan itself, so that it dynamically changes the "inactive" color based on the current color scheme.

**Comment by giantclambake** [comment]
e42d6f4  would appear to have fixed this ~ thanks ;)
